core/src/main/java/org/bitcoinj/wallet/WalletProtobufSerializer.java

```diff
@@ -114,14 +116,27 @@ public void setRequireMandatoryExtensions(boolean value) {
         requireMandatoryExtensions = value;
     }
 
+	/**
+     * Change buffer size for writing wallet to output stream. Default is {@link com.google.protobuf.CodedOutputStream.DEFAULT_BUFFER_SIZE}
+     * @param walletWriteBufferSize - buffer size in bytes
+     */
+    public void setWalletWriteBufferSize(int walletWriteBufferSize) {
+        this.walletWriteBufferSize = walletWriteBufferSize;
+    }
+
     /**
      * Formats the given wallet (transactions and keys) to the given output stream in protocol buffer format.<p>
      *
      * Equivalent to <tt>walletToProto(wallet).writeTo(output);</tt>
      */
     public void writeWallet(Wallet wallet, OutputStream output) throws IOException {
         Protos.Wallet walletProto = walletToProto(wallet);
-        walletProto.writeTo(output);
+
+        int serializedSize = walletProto.getSerializedSize();
```

Comment:
Wouldn't this forced call slow down saving considerably? Do we need the minimum calculation? It doesn't hurt to pick a larger buffer, no?